### DISCUSSION TITLE
About the "ttl" policy

### DISCUSSION THREAD

**Discussion by iventan** [risingwavelabs/risingwave#5254] [Q&A] [answered]
### Is your feature request related to a problem? Please describe.

Is "ttl" a data retention policy? Is there an estimated completion time?

**Comment by fuyufjh** [comment, marked as answer]
Yes. TTL is a data retention policy for the persisted streaming state as well as materialized views. Once TTL is enabled, the rows (key-value entries) will be invisible after a certain period of time, and will be vacuumed later by the GC process of storage. In our design, TTL is helpful for cleaning out-of-date state to keep the dataset small.

The feature is not stabilized yet. Since it may break some assumptions in the streaming engine, more testing is required to ensure reliability. 

BTW, It would be very helpful if you can explain your scenarios or requirements for this feature.

**Comment by iventan** [comment]
Looking forward to the early release of the TTL policy. thanks for your answer
